esp32c3-hal/examples/embassy_wait.rs

```diff
@@ -31,7 +31,7 @@ async fn main(_spawner: Spawner) {
         esp32c3_hal::timer::TimerGroup::new(peripherals.TIMG0, &clocks),
     );
 
-    let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
+    let io = IO::new_async(peripherals.GPIO, peripherals.IO_MUX);
```

Comment:
How does creating `IO` in a blocking way, prevent async calls on the pins produced? I guess the pins also need a type state :(.

Reply:
Maybe we just make the async handler accessible, and document then if you want to mix async pins and interrupt pins, you must make sure to call the async handler too?

Reply:
oh yes ... type state everywhere

I like the idea of letting the user mix-and-match async and blocking - on the other hand forcing users to not misuse the API at compile time feels a bit cleaner. I wonder if the freedom of mixing async and blocking is something that people would like to have.

In that case it could be nice if we could at least somehow check they are calling our async handler at runtime - no idea how to do that unfortunately